### DISCUSSION TITLE
Connecting with chatgpt4 web?

### DISCUSSION THREAD

**Discussion by eugene-so** [danielmiessler/Fabric#625] [Q&A]
I'm not sure what to do with this part of the instructions:

export OPENAI_BASE_URL=https://YOUR-SERVER:8000/v1/
export DEFAULT_MODEL="YOUR_MODEL"
And if your server needs authentication tokens, as Blablador does, you export the token the same way you would with OpenAI:

export OPENAI_API_KEY="YOUR TOKEN"

Am I supposed to setup an open AI server or something? Can I just connect it to my chatgpt account on the web?

**Comment by timothyjoh** [comment]
You don't connect to "chatGPT" the product, you use an openai developer key you get from signing up https://platform.openai.com

**Comment by AO12612** [comment]
Eugene, your confusion is understandable. 
In summary, if you plan on using a model (local or remote) that uses the openai api format, that part of the instructions is how you do it. If not  and  you plan on using openai, you need a key and those fields are automatically managed by 'fabric --setup' 

in a little more detail, as timothy said, fabric connects to api interfaces of  different models and authenticates with a "key".
Many services and some locally installable models are using the openai api interface structure as a de-facto standard.

fabric doesn't really care who or what answers on the other side. So as a "dirty" shortcut if you fill in the data of these other services in the openai fields it works transparently. This in turn means that fabric has the opportunity of interfacing with a much broader variety of providers.

That said it is immature and confusing to represent it this way in the setup files, so hopefully in the rewrite of the code these other models can be integrated more directly.
Also the documentation is really lacking given the serious amount of attention and the fantastic potential of fabric.

**Reply by eugene-so** [reply]
Thanks. Appreciate the response. Yes, I agree that fabric seems to have ton of potential. Unfortunately I'm trying to get it up and running quickly with very short windows of opportunity and I'm not very familiar with AI stuff except for chat gpt, so it is blocking my progress. Hopefully I'll have a window to try it again in the near future.
